### ISSUE TITLE
Failed installation

### ISSUE THREAD

**Issue by alejandroriglesias** [RahmanTeam/DECoN#18]
Hello,

I am trying to install DECoN on my Mac (macOS Catalina 10.15.2). I already have gcc (installed with homebrew) and gfortran seems to be included in the gcc installation. I then try to run the setup.sh, and lots of connection errors start to appear:
_* installing *source* package ‘packrat’ ...
** package ‘packrat’ successfully unpacked and MD5 sums checked
** R
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (packrat)
Warning: unable to access index for repository http://bioconductor.org/packages/3.0/bioc/bin/macosx/el-capitan/contrib/3.5:
  no fue posible abrir la URL 'http://bioconductor.org/packages/3.0/bioc/bin/macosx/el-capitan/contrib/3.5/PACKAGES'
Warning: unable to access index for repository http://bioconductor.org/packages/3.0/data/annotation/bin/macosx/el-capitan/contrib/3.5:
  no fue posible abrir la URL 'http://bioconductor.org/packages/3.0/data/annotation/bin/macosx/el-capitan/contrib/3.5/PACKAGES'
Warning: unable to access index for repository http://bioconductor.org/packages/3.0/data/experiment/bin/macosx/el-capitan/contrib/3.5:
  no fue posible abrir la URL 'http://bioconductor.org/packages/3.0/data/experiment/bin/macosx/el-capitan/contrib/3.5/PACKAGES'
Warning: unable to access index for repository http://bioconductor.org/packages/3.0/extra/bin/macosx/el-capitan/contrib/3.5:
  no fue posible abrir la URL 'http://bioconductor.org/packages/3.0/extra/bin/macosx/el-capitan/contrib/3.5/PACKAGES'_

I don't know if these warnings are important. Anyway, the program continues to execute and then jumps to the next error, which is the installation of BiocGenerics. I manually downloaded the tar.gz and skipped the error, but then another appeared.

_Installing Rcpp (0.11.6) ... 
[1]
 "Command failed (1)\n\nFailed to run system command:\n\n\t'/Library/Frameworks/R.framework/Resources/bin/R' --vanilla CMD INSTALL '/private/var/folders/3q/skwv9bz16r3gc91k9ycdzq6h0000gn/T/RtmpGpOjnz/Rcpp' --library='/Users/alex/Desktop/R/Software/DECoN-master/Linux/packrat/lib/x86_64-apple-darwin15.6.0/3.5.1' --install-tests --no-docs --no-multiarch --no-demo \n\nThe command failed with output:\n* installing *source* package 'Rcpp' ...\n** package 'Rcpp' successfully unpacked and MD5 sums checked\n** libs\nclang++  -I\"/Library/Frameworks/R.framework/Resources/include\" -DNDEBUG -I../inst/include/  -I/usr/local/include   -fPIC  -Wall -g -O2  -c Date.cpp -o Date.o\nIn file included from Date.cpp:31:\nIn file included from ../inst/include/Rcpp.h:73:\n../inst/include/Rcpp/Rmath.h:222:55: error: no member named 'Rf_pythag' in the global namespace; did you mean 'pythag'?\n    inline double pythag(double a, double b)    { return ::Rf_pythag(a, b); }\n                                                         ^~~~~~~~~~~\n                                                         pythag\n../inst/include/Rcpp/Rmath.h:222:19: note: 'pythag' declared here\n    inline double pythag(double a, double b)    { return ::Rf_pythag(a, b); }\n                  ^\n1 error generated.\nmake: *** [Date.o] Error 1\nERROR: compilation failed for package 'Rcpp'\n* removing '/Users/alex/Desktop/R/Software/DECoN-master/Linux/packrat/lib/x86_64-apple-darwin15.6.0/3.5.1/Rcpp'"
Error: Command failed (1)

Failed to run system command:

	'/Library/Frameworks/R.framework/Resources/bin/R' --vanilla CMD INSTALL '/private/var/folders/3q/skwv9bz16r3gc91k9ycdzq6h0000gn/T/RtmpGpOjnz/Rcpp' --library='/Users/alex/Desktop/R/Software/DECoN-master/Linux/packrat/lib/x86_64-apple-darwin15.6.0/3.5.1' --install-tests --no-docs --no-multiarch --no-demo 

The command failed with output:
* installing *source* package 'Rcpp' ...
** package 'Rcpp' successfully unpacked and MD5 sums checked
** libs
clang++  -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I../inst/include/  -I/usr/local/include   -fPIC  -Wall -g -O2  -c Date.cpp -o Date.o
In file included from Date.cpp:31:
In file included from ../inst/include/Rcpp.h:73:
../inst/include/Rcpp/Rmath.h:222:55: error: no member named 'Rf_pythag' in the global namespace; did you mean 'pythag'?
    inline double pythag(double a, double b)    { return ::Rf_pythag(a, b); }
                                                    
Además: Warning message:
In packrat::restore() :
  The most recent snapshot was generated using R version 3.1.2_

I am now stuck here and cannot find a solution browsing on the internet. Anyone has any ideas? I would very much like to be able to use DECoN.

Thanks in advance.

**Comment by flapa86** [comment]
I've a similar error....did you find the solution?
I've R version 3.6.0

**Comment by zgtman** [comment]
Absolutely agree. I try to install this tools with no success. I do not understand why developers do not put all necessary packages  under the packrat directory. Still with some error. I think this project is dead. I would everybody recommend rather ExomeDepth.

Thank you.

**Comment by seon22break** [comment]
> I've a similar error....did you find the solution?
> I've R version 3.6.0

The R version is very important and in a project like this, where they are not updating the code, I would recommend you to install the R 3.1.2 version. I am creating a Docker image with DeCoN installed. As soon as I have it, I will post it.

**Comment by zgtman** [comment]
Yes I did - use Exome Depth for CNV calling - works perfectly fine.

**Comment by ClaraffD** [comment]
> > I've a similar error....did you find the solution?
> > I've R version 3.6.0
> 
> The R version is very important and in a project like this, where they are not updating the code, I would recommend you to install the R 3.1.2 version. I am creating a Docker image with DeCoN installed. As soon as I have it, I will post it.

Hi, have you created your Docker image? Running out of options to successfully use DECoN here.. thank you!

**Comment by seon22break** [comment]
> > > I've a similar error....did you find the solution?
> > > I've R version 3.6.0
> > 
> > 
> > The R version is very important and in a project like this, where they are not updating the code, I would recommend you to install the R 3.1.2 version. I am creating a Docker image with DeCoN installed. As soon as I have it, I will post it.
> 
> Hi, have you created your Docker image? Running out of options to successfully use DECoN here.. thank you!

Yes , I have a docker image with DeCoN.  Works correctly

**Comment by ClaraffD** [comment]
> > > > I've a similar error....did you find the solution?
> > > > I've R version 3.6.0
> > > 
> > > 
> > > The R version is very important and in a project like this, where they are not updating the code, I would recommend you to install the R 3.1.2 version. I am creating a Docker image with DeCoN installed. As soon as I have it, I will post it.
> > 
> > 
> > Hi, have you created your Docker image? Running out of options to successfully use DECoN here.. thank you!
> 
> Yes , I have a docker image with DeCoN. Works correctly

Would it be possible for you to link it please? Thank you for your help

**Comment by worker000000** [comment]
> Yes I did - use Exome Depth for CNV calling - works perfectly fine.

do you kown which column of exomedepth stands for copy number, thanks a lot
and have you ever met this https://github.com/vplagnol/ExomeDepth/issues/44

**Comment by worker000000** [comment]
@ClaraffD 
i once download one, not writen by me, but it needs to enter the docker, can not concatate argument directly, not that convenient

#################################################################
# Dockerfile # # Version: 0.1
# Software: DECoN V1.0.2
# Description: DECoN Detection of Exon Copy Number variants
# Website: http://www.thetgmi.org/tgmi-news/detecting-tricky-gene-mutations/
# Tags: None, for the moment
# Base Image: r-base:3.1.2
#################################################################
From r-base:3.1.2
ENTRYPOINT ["/bin/bash"]
### Adding user decon
RUN useradd -ms /bin/bash decon
### Adding Certificates for wget tu download from git
RUN apt-get update && \
    apt-get install -y --no-install-recommends \
    ca-certificates \
    && rm -rf /var/lib/apt/lists/*
### Running Container as decon and install decon from git
USER decon
WORKDIR /home/decon/
RUN wget https://github.com/RahmanTeam/DECoN/archive/v1.0.2.tar.gz
RUN tar -xzvf v1.0.2.tar.gz
Run rm v1.0.2.tar.gz
WORKDIR /home/decon/DECoN-1.0.2/Linux/
### Starting setup
RUN mkdir /home/decon/DECoN-1.0.2/Linux/packrat/src/VGAM/
RUN wget https://cran.r-project.org/src/contrib/Archive/VGAM/VGAM_0.9-8.tar.gz -O /home/decon/DECoN-1.0.2/Linux/packrat/src/VGAM/VGAM_0.9-8.tar.gz
RUN ./setup.sh
